src/main_thread/text_displayer/html/__tests__/utils.test.ts

```diff
@@ -333,3 +334,50 @@ describe("HTML Text buffer utils - removeCuesInfosBetween", () => {
     ]);
   });
 });
+
+describe("HTML areCuesStartNearlyEqual", () => {
+  it("Case 1: should be false", () => {
+    // * [0, 2] and [2, 4] start are NOT equals
+    const element = document.createElement("div");
+    const cues = [{ start: 1, end: 2, element }];
+    const cueInfo1 = { start: 0, end: 2, cues };
+    const cueInfo2 = { start: 2, end: 4, cues };
+
+    expect(areCuesStartNearlyEqual(cueInfo1, cueInfo2)).toBe(false);
+  });
+  it("Case 2: should be true", () => {
+    // * [0, 2] and [0, 4]  start are equals
+    const element = document.createElement("div");
+    const cues = [{ start: 1, end: 2, element }];
+    const cueInfo1 = { start: 0, end: 2, cues };
+    const cueInfo2 = { start: 0, end: 4, cues };
+    expect(areCuesStartNearlyEqual(cueInfo1, cueInfo2)).toBe(true);
+  });
+
+  it("Case 3: should be false", () => {
+    // * [0, 0.1] and [0.101, 2] start are NOT equals
+    const element = document.createElement("div");
+    const cues = [{ start: 1, end: 2, element }];
+    const cueInfo1 = { start: 0, end: 0.1, cues };
+    const cueInfo2 = { start: 0.101, end: 2, cues };
+    expect(areCuesStartNearlyEqual(cueInfo1, cueInfo2)).toBe(false);
+  });
+
+  it("Case 4: should be true", () => {
+    // * [0, 2] and [0.01, 4]  start are equals
+    const element = document.createElement("div");
+    const cues = [{ start: 1, end: 2, element }];
+    const cueInfo1 = { start: 0, end: 2, cues };
+    const cueInfo2 = { start: 0.01, end: 4, cues };
+    expect(areCuesStartNearlyEqual(cueInfo1, cueInfo2)).toBe(true);
+  });
+
+  it("Case 5: should be false", () => {
+    // * [0, 100] and [1, 200]  start are NOT equals
+    const element = document.createElement("div");
+    const cues = [{ start: 1, end: 2, element }];
+    const cueInfo1 = { start: 0, end: 100, cues };
+    const cueInfo2 = { start: 2, end: 200, cues };
+    expect(areCuesStartNearlyEqual(cueInfo1, cueInfo2)).toBe(false);
+  });
```

Comment:
It misses test cases for what constitutes "nearly" equal no? Just to make it a proper unit test for the use case